thriftbp/errors.go

```diff
@@ -0,0 +1,89 @@
+package thriftbp
+
+import (
+	"errors"
+
+	retry "github.com/avast/retry-go"
+	baseplatethrift "github.com/reddit/baseplate.go/internal/gen-go/reddit/baseplate"
+	"github.com/reddit/baseplate.go/retrybp"
+)
+
+// BaseplateErrorCode defines the minimum interface for a Baseplate Error that is
+// used by logic within baseplate.go.
+//
+// Additional fields are defined in BaseplateErrorDetails.
+type BaseplateErrorCode interface {
```

Comment:
:no_bell: I still don't see the value of exporting those interfaces. service code (users of this library) don't really need them because they will have the thrift compiled go code, which has the  concrete types in them.

Reply:
I can be useful for services that are interacting with multiple thrift services and what to be able to write helper methods for dealing with Baseplate errors. Otherwise they have a separate “Error” type for each service

Reply:
even they have multiple thrift files, those should include the same `baseplate.thrift`, so there should be only one compiled error strict ?